samples/examples/FSharpExamples/Matrix.fs

```diff
@@ -28,38 +28,53 @@ open Prajna.Api.FSharp
 
 open Prajna.Examples.Common
 
-type MatrixSample() =
+module Matrix =
 
     let (.*) (xs: float[]) (ys: float[]) = 
-        // simulate doing some more work, 
-        // e.g.: run the remaining layers of a neural net
-        Threading.Thread.Sleep(3) 
         Array.fold2 (fun sum x y -> sum + x * y) 0.0 xs ys 
 
     let printTiming (msg: string) (x: Lazy<'b>) : 'b =
         let sw = Stopwatch.StartNew()
+        printf "%s: " msg
         let ret = x.Value
         sw.Stop()
-        printfn "%s: %A" msg sw.Elapsed
+        printfn "%A" sw.Elapsed
         ret
 
-    let run (cluster: Cluster) =
+    let random = new Random(0)
+
+    let randomVector n =  Array.init n (fun _ -> random.NextDouble())
+    
+    let randomMatrix m n =  Array.init m (fun _ -> randomVector n)
+
+    /// Matrix-Vector multiplication for illustrational purposes.
+    /// The overhead of distributing makes it faster to run this locally.
+    /// See Matrix-Matrix for a case where the computation is actually faster distributed.
+    let matrixVector (cluster: Cluster) =
 
         let numRows = 200
         let vectorSize = 100
 
-        let r = new Random(0)
-        let randomMatrix =  Array.init numRows (fun _ -> Array.init vectorSize (fun _ -> r.NextDouble()))
-        let vector = Array.init vectorSize (fun _ -> r.NextDouble())
+        let matrix =  randomMatrix numRows vectorSize
+        let vector = randomVector vectorSize
 
         let singleCoreResult = 
             printTiming "single core" <| lazy 
-                randomMatrix |> Array.map (fun row -> row .* vector)
-        
+                matrix |> Array.map (fun row -> row .* vector)
+
+        let numPartitions = Environment.ProcessorCount
```

Comment:
It can be thought in a different way. A partition is a unit for scheduling and failure recovery. The proper size of a partition can often be determined by the application itself instead of # of nodes. Usually, # of partitions can be larger or even far larger than # of nodes (or total # of processors).